qemu/tests/rh_qemu_update.py

```diff
@@ -0,0 +1,265 @@
+    6) Verify installed qemu-kvm version
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    qemu_package = params['qemu_package_install']
+
+    def install_qemu(session):
+        """
+        Install compose version of qemu-kvm pkg by the name provided in cfg
+        """
+        cmd = "{} install -y {}".format(
+            params['pm_tool'], qemu_package
+        )
+        s, o = session.cmd_status_output(cmd, timeout=OPERATION_TIMEOUT)
+        if s != 0:
+            test.error("Installation of '{}' failed with: {}".format(
+                qemu_package, o
+                )
+            )
+        test.log.info("Installation of '{}' succeeded".format(
+            qemu_package
+            )
+        )
+
+    def install_component_management(session):
+        """
+        Clone component management repository for preparation of necessary
+        repositories containing qemu-kvm build or whole virt module
+        """
+        cmd_clone = "git clone --depth=1 {} -b {} {}".format(
+            params['cm_repo'],
+            params['cm_branch'],
+            params['cm_path']
+        )
+        s_clone, o_clone = session.cmd_status_output(
+            cmd_clone, timeout=OPERATION_TIMEOUT
+            )
+        if s_clone != 0:
+            test.error("Clonning of '{}' failed with: {}".format(
+                params['cm_repo'], o_clone
+                )
+            )
+        test.log.info("Clonning of '{}' succeeded".format(
+            params['cm_repo']
+            )
+        )
+
+    def _get_installed_qemu_info(session=None):
+        """
+        Get info about qemu versions used on guest or host, function returns
+        dict containing NVR, TARGET and module_id (in case of module only)
+        """
+        cmd = "rpm -q qemu-kvm"
+        if session is not None:
+            out = session.cmd(cmd, timeout=QUERY_TIMEOUT)
+            tgt = process.run(
+                "cat /etc/os-release | grep VERSION_ID | cut -d'=' -f2",
+                shell=True
+            ).stdout_text.strip().replace("\"", "")
+        else:
+            out = process.run(cmd, shell=True).stdout_text.strip()
+            distro_details = distro.detect()
+            tgt = f"{distro_details.version}.{distro_details.release}"
+        # Drop arch information from NVR e.g. '.x86_64'
+        nvr = out.rsplit('.', 1)[0]
+        # Expect module is used if nvr contains '+' character
+        if "+" in nvr:
+            repo_module = process.run(
+                f"grep `{params['pm_tool']} info {params['qemu_package_install']} | " +
+                "grep 'From repo' | cut -d'-' -f2` /etc/yum.repos.d/* | head -1",
+                shell=True
+            ).stdout_text
+            return {
+                "nvr": nvr,
+                "target": tgt,
+                "module_id": re.search("[0-9]{5}", repo_module)[0]
```

Comment:
Before, module_id is used to get module stream from mbs, and after our discussion offline, we use the enabled module stream of host,  the nvr is enough for compare version, create repo.  So can you explain the reason to keep using module id ?   Is there anything i missed ? Thanks!

Reply:
No, no, you're completely right. You didn't miss anything. Actually it's me who started mixing things. 

So I re-worked my PR, entirely dropped module_id usage and changed the way we detect module (via "+").